tools/test-emails

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import os
+import sys
+import django
+from django.template import loader, TemplateDoesNotExist
+
+print("Starting...")
+ZULIP_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..')
+OUTPUT_PATH = os.path.join(ZULIP_PATH, 'static', 'generated', 'test-emails.html')
+sys.path.append(ZULIP_PATH)
+os.environ['DJANGO_SETTINGS_MODULE'] = 'zproject.settings'
+django.setup()
+
+def render_email(template_prefix, context):
+    # type: (str, dict) -> str
+    print("Rendering template " + template_prefix)
+    subject = loader.render_to_string('zerver/emails/' + template_prefix + '.subject', context).strip()
+    message = loader.render_to_string('zerver/emails/' + template_prefix + '.txt', context)
+
+    try:
+        html_message = loader.render_to_string('/zerver/emails/' + template_prefix + '.html', context)
+    except TemplateDoesNotExist:
+        html_message = None
+
+    info = '<h3>Test for {}</h3><h4>Subject: {}</h4><br/>'.format(template_prefix, subject)
+    if html_message is not None:
+        return info + html_message
+
+    message = message.replace('<', '&lt;').replace('>', '&gt;')
+    return '{}<pre>{}</pre>'.format(info, message)
+
+# write fake data for all variables
+test_context = {
+    'user_profile': {
+        'full_name': 'Zooly Zulipson',
+        'email': 'zooly@zulip.org',
+        'realm': {
+            'uri': 'https://chat.zulip.org',
+            'name': 'Zulip Community',
+        },
+    },
+    'realm': {
+        'uri': 'https://chat.zulip.org',
+        'name': 'Zulip Community',
+    },
+    'device_info': {
+        'login_time': "12/12/12, 12:12:12",
+        'device_browser': "Firefox",
+        'device_os': "ZulipOS",
+        'device_ip': '127.0.0.1',
+    },
+    'referrer': {
+        'full_name': 'Zooly Zulipson',
+        'email': 'zooly@zulip.org',
+    },
+    'user': {
+        'full_name': 'Zooly Zulipson',
+        'email': 'zooly@zulip.org',
+    },
+    'referrer_name': 'Zooly Zulipson',
+    'referrer_email': 'Zooly Zulipson',
+    'realm_uri': 'https://chat.zulip.org',
+    'server_uri': 'https://chat.zulip.org',
+    'old_email': 'oldmail@zulip.org',
+    'new_email': 'newmail@zulip.org',
+    'activate_url': 'https://test.zulip.org',
+    'support_email': 'admin@zulip.org',
+    'zulip_support': 'admin@zulip.org',
+    'unsubscribe_link': 'https://unsub.zulip.org',
+}
+
+paths = [
```

Comment:
Maybe we can add some error handling in here to be able to handle the crash and spit out appropriate message.